src/utils/cloud_storage.R

```diff
@@ -0,0 +1,181 @@
+#' uploaded to the bucket.
+#'
+#' @param df Data frame to save out.
+#' @param name Name of the file to read, including prefix (`input/` or `output/`)
+#'     and filetype `.parquet`.
+#'
+#' @returns Nothing, but file is saved to the `global-monitoring` bucket.
+#'
+#' @export
+update_gcs_file <- function(df, name) {
+  tf <- tempfile(fileext = ".parquet")
+  arrow$write_parquet(
+    x = df,
+    sink = tf
+  )
+
+  if (gmas_test_run()) {
+    message(
+      "`update_pq_file()` not saving data as `gmas_test_run()` is `TRUE`. ",
+      "Set `GMAS_TEST_RUN` env variable to `FALSE` if you want the data to be ",
+      "saved, but be careful of sending emails or calling the OpenAI API."
+    )
+    return(invisible(NULL))
+  }
+
+  invisible(
+    gcs$gcs_upload(
+      file = tf,
+      bucket = "global-monitoring",
+      name = name,
+      predefinedAcl = "bucketLevel"
+    )
+  )
+}
+
+#' Save data frame to Google Sheet
+#'
+#' DEPRECATED: This only exists to save out files for CERF, but will eventually
+#' be removed once CERF switches to API access.
+#'
+#' A convenient file saver that saves the data frame to the specified
+#' name. Performs the same file name searching with ^ and $ as in `read_gs_file()`
+#' then simply writes out the data frame using [googlesheets4::write_sheet()].
+#' The `sheet` name is specified to explicitly be `name`, matching `read_gs_file()`.
+#'
+#' @param df Data frame to save out.
+#' @param name Name of the Google Sheet.
+#'
+#' @export
+update_gs_file <- function(df, name) {
+  ss <- get_gs_file(paste0("^", name, "$"))
+  if (gmas_test_run()) {
+    message(
+      "`update_gs_file()` not saving data as `gmas_test_run()` is `TRUE`. ",
+      "Set `GMAS_TEST_RUN` env variable to `FALSE` if you want the data to be ",
+      "saved, but be careful of sending emails or calling the OpenAI API."
+    )
+    return(invisible(NULL))
+  }
+  googlesheets4$write_sheet(
+    data = df,
+    ss = ss,
+    sheet = name
+  )
+}
+
+#' Find GCS file names matching pattern
+#'
+#' Pulls names from the `global-monitoring` bucket and
+#' filters those names by the passed patterns, if passed. If `pattern` is `NULL`,
+#' then all files in the bucket are returned.
+#'
+#' @param pattern Pattern to look for. Passed to [stringr::str_detect()]
+#'
+#' @export
+gcs_file_detect <- function(pattern = NULL) {
+  file_names <- gcs$gcs_list_objects(bucket = "global-monitoring")$name
+  if (!is.null(pattern)) {
+    file_names <- file_names[stringr$str_detect(file_names, pattern)]
+  }
+  file_names
+}
+
+##############################
+#### SETUP AUTHORIZATIONS ####
+##############################
+
+# authorize drive access
+# TODO: Drop Google Sheets and Drive
+googlesheets4$gs4_auth(
+  path = Sys.getenv("GLOBAL_MONITORING_JSON")
+)
+
+googledrive$drive_auth(
+  path = Sys.getenv("GLOBAL_MONITORING_JSON")
+)
+
+gcs$gcs_auth(
+  json_file = Sys.getenv("GLOBAL_MONITORING_JSON")
+)
```

Comment:
maybe a separate file for these authorizations to make it easier for newcomers to navigate file structure (improve long term maintenance)?

Could also separate deprecated gdrive functions (same purpose i think)